tests/quantization/gptq/test_gptq.py

```diff
@@ -242,12 +244,11 @@ def test_change_loading_attributes(self):
         with tempfile.TemporaryDirectory() as tmpdirname:
             self.quantized_model.save_pretrained(tmpdirname)
             if not self.use_exllama:
-                self.assertEqual(self.quantized_model.config.quantization_config.use_exllama, False)
+                self.check_quantized_layers_type(self.quantized_model, "cuda-old")
                 # we need to put it directly to the gpu. Otherwise, we won't be able to initialize the exllama kernel
                 quantized_model_from_saved = AutoModelForCausalLM.from_pretrained(
                     tmpdirname, quantization_config=GPTQConfig(use_exllama=True, bits=4), device_map={"": 0}
                 )
-                self.assertEqual(quantized_model_from_saved.config.quantization_config.use_exllama, True)
```

Comment:
Why remove this line?

Reply:
With this [PR](https://github.com/huggingface/optimum/pull/1574/files#), we don't save all the arguments anymore (only those in `self.serialization_keys`) by modifying `to_dict()` . The issue with that is that we are updating the quantization_config based on the one from optimum: `config.quantization_config = GPTQConfig.from_dict_optimum(quantizer.to_dict())`. 
This line was needed since some args could change like `use_exllama` in optimum. 

I was thinking on doing a PR to remove this line, and maybe not save args related to inference anymore (use_exllama,...) or revert the PR on optimum. What are your thoughts ?

Reply:
> This line was needed since some args could change like use_exllama in optimum.

Sorry, I don't completely follow. Does this mean that `use_exllama` will no longer change and the test check is no longer required. 

> I was thinking on doing a PR to remove this line, and maybe not save args related to inference anymore (use_exllama,...) 

It depends. This can be considered a breaking change, as users might now expect these values in their configs. The most important thing is for old configs to still be loadable and produce the same result.

Reply:
> Sorry, I don't completely follow. Does this mean that use_exllama will no longer change and the test check is no longer required?

Basically, I mean that the user can set `use_exllama=True` in transformers and this value can change in optimum (`use_exllama=False`). However, since we don't serialize it anymore in optimum GPTQ config, `use_exllama` will be set to the default value through: `config.quantization_config = GPTQConfig.from_dict_optimum(quantizer.to_dict()).`

> It depends. This can be considered a breaking change, as users might now expect these values in their configs. The most important thing is for old configs to still be loadable and produce the same result.

Yes, the old configs will still work. However, for new users, they will have to pass these args each time. 

I will probably work on the second option then since from the start, we should not have to let the user select the kernel since we can switch from one to another.